test/mw_shim.js

```diff
@@ -0,0 +1,849 @@
+/**
+ * Helper functions to fake mw object data for CLI testing of the Morebits library.
+ *
+ * Includes just what we care about:
+ * log
+ * Map
+ * config (barebones, will need data set)
+ * util (but plenty of excessive functions)
+ *     - All of morebits just needs: escapeRegExp, getParamValue, getUrl, isIPv6Address, wikiScript
+ *         - getParamValue (mb.w.page), wikiScript (mb.w.api), getUrl (batch, others)
+ *         - getUrl needs wikiScript, wikiUrlEncode, escapeIdForLink
+ *     - twinkle.js would need: addPortletLink
+ *     - modules would need: addCSS, isIPAddress, rawurlencode, wikiUrlencode
+ *
+ * Not funtional/done:
+ * user.tokens (nonfunctional, only used in Morebits.wiki.page)
+ * Title (only used in Morebits.wiki.page)
+ */
+
+var mw = {};
+
+/**
+ * mw.log
+ * We just need the `warn` method, but might as well mw.log = console.log
+ * node repl doesn't have console.warn
+ * https://doc.wikimedia.org/mediawiki-core/master/js/source/mediawiki.html#mw-log-method-warn
+ */
+mw.log = console && Function.prototype.bind.call(console.log, console);
+mw.log.warn = console && console.warn ?
+	Function.prototype.bind.call(console.warn, console) :
+	function(){};
+
+/**
+ * mw.Map
+ * Needed for various things, in particular mw.config
+ * https://doc.wikimedia.org/mediawiki-core/master/js/source/mediawiki.html#mw-Map-method-constructor
```

Comment:
These are not stable links for source code.

Use something like <https://gerrit.wikimedia.org/g/mediawiki/core/+/1.35.0/resources/src/startup/mediawiki.js> or <https://github.com/wikimedia/mediawiki/blob/ea586cfa2e/resources/src/startup/mediawiki.js> instead.

When browsing GitHub, you can press `y` to get a stable permalink.